test/functional/update_v2/update-client-certificate.bats

```diff
@@ -0,0 +1,111 @@
+#!/usr/bin/env bats
+
+load "../testlib"
+
+server_pub="/tmp/$TEST_NAME-server-pub.pem"
+server_key="/tmp/$TEST_NAME-server-key.pem"
+client_pub="/tmp/$TEST_NAME-client-pub.pem"
+client_key="/tmp/$TEST_NAME-client-key.pem"
+
+global_setup() {
+
+	# Skip this test for local development because it takes a long time. To run this
+	# test locally, configure swupd with --fallback-ca-paths=/tmp/swup_test_certificates
+	# and run: TRAVIS=true make check
+	if [ -z "${TRAVIS}" ]; then
+		return
+	fi
+
+	create_test_environment "$TEST_NAME"
+	create_test_environment "$TEST_NAME" 100
+
+	create_bundle -n test-bundle -f /usr/bin/test-file "$TEST_NAME"
+
+	run sudo sh -c "mkdir -p $CLIENT_CERT_DIR"
```

Comment:
From my understanding of bats, the run helper is intended to be used only to run the command under test, I've noticed you use it sometimes for setup steps, I think in those cases you should use a regular shell command and leave "run" only for the command to be tested.